src/main/java/spring/intro/config/AppConfig.java

```diff
@@ -3,26 +3,32 @@
 import java.util.Properties;
 import javax.sql.DataSource;
 import org.apache.commons.dbcp2.BasicDataSource;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
 import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
 import spring.intro.model.User;
 
 @Configuration
+@PropertySource("classpath:db.properties")
 @ComponentScan(basePackages = {
         "spring.intro.service",
         "spring.intro.dao"
 })
 public class AppConfig {
+    @Autowired
+    private Environment environment;
```

Comment:
We need to use constructor injection to improve the fault-tolerance of the application (remember that sometimes field injection can fail).

Reply:
You absolutely rigth, thanks